### ISSUE TITLE
bun-types should be @types/bun as it disables @types/* as of 1.0.7

### ISSUE THREAD

**Issue by wizzard0** [oven-sh/bun#6700]
### TLDR:

Please re-publish `bun-types` as `@types/bun`, remove creation of `types` field in `tsconfig.json` and a ton of import errors will go away.

### What version of Bun is running?

1.0.7

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

1. `bun init` 
2.  try to mention any DOM type
3.  `tsc`

### What is the expected behavior?

"@types/bun" package is installed, "types" section of the tsconfig.json is not modified.
Running `tsc` results in `Found 0 errors.`

### What do you see instead?

Multiple `error TS2304: Cannot find name '...'`

Why: The recommended `"types": ["bun-types"]` settings in Bun-generated tsconfig.json [overrides and disables](https://www.typescriptlang.org/tsconfig#types) search for the typings in `@types/*` altogether and forces to explicitly list everything unless you do
```
    "types": [
      "typescript/lib/lib.dom.d.ts",
      "bun-types",
      "@types/your-dependency",
      ...
    ]
```

Since you have to list every typing import you use manually, it would be great if Bun could play nice with the default `tsc` conventions.

### Additional information

This change will also fix #5983, #5134, #4980, #3030, #463 and other similar issues

**Comment by Electroid** [comment]
Duplicate of #5674

**Comment by wizzard0** [comment]
@Electroid re: not planned: why? are there any obvious downsides?

**Comment by Electroid** [comment]
@wizzard0 We plan to do this. Unfortunately, GitHub labels duplicate issues that are closed as "not planned".

**Comment by wizzard0** [comment]
Ah, OK
